Sources/PublicInterface/Client.swift

```diff
@@ -228,16 +237,34 @@ public class Client: WalletConnect {
 
     private func expectUpdateSessionRequest(_ request: Request) {
         if request.method == "wc_sessionUpdate" {
-            guard let approval = sessionApproval(from: request) else {
+            guard let info = sessionInfo(from: request) else {
                 // TODO: error handling
                 try! send(Response(request: request, error: .invalidJSON))
                 return
             }
-            guard let session = communicator.session(by: request.url), !approval else { return }
-            do {
-                try disconnect(from: session)
-            } catch { // session already disconnected
-                delegate.client(self, didDisconnect: session)
+            guard let session = communicator.session(by: request.url) else { return }
+            if !info.approved {
+                do {
+                    try disconnect(from: session)
+                } catch { // session already disconnected
+                    delegate?.client(self, didDisconnect: session)
+                }
+            } else {
+                if let walletInfo = session.walletInfo {
+                    let updatedInfo = Session.WalletInfo(
+                        approved: info.approved,
+                        accounts: info.accounts ?? [],
+                        chainId: info.chainId,
+                        peerId: walletInfo.peerId,
+                        peerMeta: walletInfo.peerMeta
+                    )
+                    var updatedSesson = session
+                    updatedSesson.walletInfo = updatedInfo
+                    communicator.addSession(updatedSesson)
+                    delegate?.client(self, didUpdate: updatedSesson)
+                } else {
+                    delegate?.client(self, didUpdate: session)
```

Comment:
I think this one should never happen by design. We only add sessions containing WalletInfo